File: DistroKid.Infrastructure/Errors/ErrorCodes.cs
﻿using System.Text.Json.Serialization;

namespace DistroKid.Infrastructure.Errors;

/// <summary>
/// This enumeration represents codes for common errors and should be used to better identify the error by the client. You may add or remove codes as you see fit.
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ErrorCodes
{
    Unknown,
    TechnicalError,
    EntityNotFound,
    PhysicalFileNotFound,
    UserAlreadyExists,
    WrongPassword,
    CannotRead,
    CannotAdd,
    CannotUpdate,
    CannotDelete,
    MailSendFailed,
    UserNotArtist
}

File: DistroKid.Client/src/infrastructure/apis/client/models/ErrorCodes.ts



















export const ErrorCodes = {
    Unknown: 'Unknown',
    TechnicalError: 'TechnicalError',
    EntityNotFound: 'EntityNotFound',
    PhysicalFileNotFound: 'PhysicalFileNotFound',
    UserAlreadyExists: 'UserAlreadyExists',
    WrongPassword: 'WrongPassword',
    CannotAdd: 'CannotAdd',
    CannotUpdate: 'CannotUpdate',
    CannotDelete: 'CannotDelete',
    MailSendFailed: 'MailSendFailed',
    UserNotArtist: 'UserNotArtist'
} as const;
export type ErrorCodes = typeof ErrorCodes[keyof typeof ErrorCodes];


export function instanceOfErrorCodes(value: any): boolean {
    for (const key in ErrorCodes) {
        if (Object.prototype.hasOwnProperty.call(ErrorCodes, key)) {
            if (ErrorCodes[key as keyof typeof ErrorCodes] === value) {
                return true;
            }
        }
    }
    return false;
}

export function ErrorCodesFromJSON(json: any): ErrorCodes {
    return ErrorCodesFromJSONTyped(json, false);
}

export function ErrorCodesFromJSONTyped(json: any, ignoreDiscriminator: boolean): ErrorCodes {
    return json as ErrorCodes;
}

export function ErrorCodesToJSON(value?: ErrorCodes | null): any {
    return value as any;
}

export function ErrorCodesToJSONTyped(value: any, ignoreDiscriminator: boolean): ErrorCodes {
    return value as ErrorCodes;
}

